gcapi/gcapi.py

```diff
@@ -618,8 +654,24 @@ def upload_cases(
                 filename=filename,
             )
 
-        self.raw_image_upload_sessions.process_images(
-            pk=raw_image_upload_session["pk"], json=upload_session_data
-        )
+        if process:
+            self.raw_image_upload_sessions.process_images(
+                pk=raw_image_upload_session["pk"], json=upload_session_data
+            )
 
         return raw_image_upload_session
+
+    def _get_algorithm(self, algorithm_name: str) -> dict:
+        """Get the algorithm for the given algorithm name. """
+        algorithms = [
+            a
+            for a in list(self.algorithms.list()["results"])
+            if a["title"] == algorithm_name
+        ]
```

Comment:
`algorithm_name` -> `algorithm`, and add a filter as this will fetch all algorithms the user has access to from the API, rather than 1. You want `/api/v1/algorithms/?slug=blah`, should return 1 result.